packages/cardpay-reward-programs/cardpay_reward_programs/main.py

```diff
@@ -49,11 +35,11 @@ def run_reward_program(
     """
     with open(AnyPath(parameters_file), "r") as stream:
         parameters = json.load(stream)
-    rule = to_rule(parameters)
-    start_block = rule.start_block
-    end_block = rule.end_block
-    results_df = rule.run(start_block, end_block)
-    payment_list = rule.df_to_payment_list(results_df, payment_cycle, reward_program_id)
+    
+    for subclass in Rule.__subclasses__():
+        if subclass.__name__ == rule_name:
+            rule = subclass(parameters["core"], parameters["user_defined"])
+    payment_list = rule.run(parameters["run"]["payment_cycle"], parameters["run"]["reward_program_id"])
```

Comment:
I'm still unsure:
- if the `run` key is actually available after `run_reward_program` or before `run_reward_program`. I have always thought `run` key is added only after `run_reward_program` as a record to s3.

I'll leave this to you since it's highly dependent on how the scheduler works.

Reply:
I think this is before - when we get to any of the code in this package running, it's for a specific known payment cycle and for one reward program.